Documentation/MyFirstContribution

```diff
@@ -0,0 +1,584 @@
+GitGitGadget. It may be useful to describe a workflow using git send-email as
+well.
+////
+
+=== Sending a PR to GitGitGadget
+
+GitGitGadget is a tool created by Johannes Schindelin to make life as a Git
+contributor easier for those used to the GitHub PR workflow. It's documented
+at gitgitgadget.github.io. In order to have your code tested and formatted for
+review, you need to start by opening a Pull Request against gitgitgadget/git.
+Head to https://github.com/gitgitgadget/git and open a PR either with the "New
+pull request" button or the convenient "Compare & pull request" button that may
+appear with the name of your newly pushed branch.
+
+Review the PR's title and description, as it's used by GitGitGadget as the cover
+letter for your change. When you're happy, submit your pull request.
+
+=== Getting CI to Run
+
+If it's your first time using GitGitGadget (which is likely, as you're using
+this tutorial) then someone will need to give you permission to use the tool.
+As mentioned in the GitGitGadget doc, you just need someone who already uses it
+to comment on your PR with `/allow <username>`.
+
+Once you're whitelisted, GitGitGadget will automatically run your PRs through
+the CI.
+
+If the CI fails, you can update your changes with `rebase -i` and push your
+branch again:
+
+----
+git push -f remotename psuh
+----
+
+In fact, you should continue to make changes this way up until the point when
+your patch is accepted into `next`.
+
+////
+TODO https://github.com/gitgitgadget/gitgitgadget/issues/83
+It'd be nice to be able to verify that the patch looks good before sending it
+to everyone on Git mailing list.
+=== Check Your Work 
+////
+
+=== Sending Your Patches
+
+Now that your CI is passing, sending out for review with GGG is as simple as
+commenting on your PR with `/submit`.
+
+=== Updating With Comments
+
+As documented on the GitGitGadget site, when a reviewer asks you for changes,
+you can make them using `git rebase -i`. When you're ready, force push to your
+fork's branch again, just like when you were getting the CI to pass above.
+
+////
+This is the path for git send-email. Do we want to cover that approach as well?
+=== Running with Travis On Your Fork
+
+== Sending For Review
+
+=== Preparing initial patchset
+
+=== Preparing email
+
+=== Sending email
+
+=== Applying Changes
+
+=== Sending v2
+
+End of path for git send-email
+////
+
+== Now What?
+
+Your code has been lauded by the upstream reviewers. It is perfect. It is ready
+to be accepted. You don't need to do anything else; the maintainer will pull
+your patchset into `next` and life is good.
+
+However, if it isn't so perfect, once it is in `next`, you can no longer modify
+your commits in GitGitGadget. Consider that PR "closed" - you will need to
+repeat the entire process for any bug fix commits you need to send, basing your
+changes on the maintainer's topic branch for your work instead of `master`.
```

Comment:
Mention that those topic branches are

1. Pushed to https://github.com/gitster/git
2. Mirrored to https://github.com/gitgitgadget/git
3. Mentioned in the bi-weekly "What's cooking" mails to the Git mailing list that serve as status report about those topic branches

maybe?

Reply:
Since it's mirrored to the GGG repo, can you still send patches via GGG the same way while basing them on topic branches? Meant to ask this last week and didn't get a chance.

Reply:
> Since it's mirrored to the GGG repo, can you still send patches via GGG the same way while basing them on topic branches?

Yes, you just have to be careful to use that target branch in the PR (either when you open it, or when you realize that you targeted `master` by mistake and then change it).

Reply:
Thanks!